### PR TITLE
Remove github secrets requirements

### PR DESCRIPTION
**Pull request by EngHabu**
# TL;DR
Remove the need to access github secrets on PRs to help external contributors validate their changes on PRs.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Comment by kumare3** [comment]
so we have stopped pushing the image to docker registry?

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -4,68 +4,67 @@ on:
...
     steps:
-      - uses: actions/checkout@v2
-      - name: Push Docker Image to Github Registry
-        uses: whoan/docker-build-with-cache-action@v5
+      - name: Checkout
+        uses: actions/checkout@v2
+      - id: load-docker-cache
+        uses: actions/cache@v1
+        with:
+          path: /tmp//tmp/docker-images
```
Comment (katrogan) [inline .github/workflows/pull_request.yml:15]: why the double /tmp/?

**Comment by EngHabu** [comment]
See, I really needed the 'default to draft' option :) 
Still working on it... 

RE: Ketan: Yes for pull requests we do not need to... this will enable PRs from forks to run fine too... and once we merge to master, we will still push to dockerhub & github registries...

RE: Katrina! thank you!... fixed and still iterating... I don't think caching (for docker steps) is working in propeller repo neither... so I'm iterating on that...

File: boilerplate/lyft/end2end/end2end.sh [hunk 2]
```diff
@@ -13,21 +13,17 @@ OUT="${DIR}/tmp"
 rm -rf ${OUT}
 git clone https://github.com/lyft/flyte.git "${OUT}"
 
-echo "Loading github docker images into 'kind' cluster to workaround this issue: https://github.com/containerd/containerd/issues/3291#issuecomment-631746985"
```
Comment (katrogan) [inline boilerplate/lyft/end2end/end2end.sh:16]: is this no longer applicable? i see we're still using kind, right?

File: boilerplate/lyft/github_workflows/pull_request.yml [hunk 3]
```diff
@@ -4,47 +4,82 @@ on:
   pull_request
 
 jobs:
-  build-and-end2end:
+  build-docker:
```
Comment (katrogan) [inline boilerplate/lyft/github_workflows/pull_request.yml:7]: will we update boilerplate too, eventually?

**Review by katrogan — DISMISSED** [review]
this is awesome!

File: boilerplate/lyft/end2end/end2end.sh (hunk 2 above)
Comment (EngHabu) [reply boilerplate/lyft/end2end/end2end.sh:16]: We are using kind but that specific issue was related to pulling anonymous images from github docker registry... which is the part I got rid of

File: boilerplate/lyft/github_workflows/pull_request.yml (hunk 3 above)
Comment (EngHabu) [reply boilerplate/lyft/github_workflows/pull_request.yml:7]: You mean the repo, right? yeah I've been trying to make the changes here that I can just copy paste into the repo... but so far this won't work directly... need more brain power

**Review by EngHabu — COMMENTED** [review]
I also realized I probably need to update master.yml too... given that I've updated some shared scripts... so let me review that and I'll let you if it needs another round :) thanks for your help!

File: boilerplate/lyft/end2end/end2end.sh (hunk 2 above)
Comment (katrogan) [reply boilerplate/lyft/end2end/end2end.sh:16]: thanks for explaining!

**Review by katrogan — APPROVED** [review]

**Pull request merged**